geopandas/tests/test_geodataframe.py

```diff
@@ -51,6 +51,18 @@ def test_df_init(self):
         assert type(self.df2) is GeoDataFrame
         assert self.df2.crs == self.crs
 
+    def test_df_init_repeat_geo_col(self):
+        df3 = pd.DataFrame(self.df2)
+        df3["geom"] = df3["geometry"]
+        # explicitly prevent construction of gdf with repeat geometry column names
+        # two columns called "geometry", geom col inferred
+        with pytest.raises(ValueError):
+            GeoDataFrame(df3.rename(columns={"geom": "geometry"}))
```

Comment:
Can you do the rename as a separate step before the pytest.raises? (to make it clearer that it's the GeoDataFrame(..) that should raise the error). And same below.